pysm3/models/dust_layers.py

```diff
@@ -0,0 +1,150 @@
+        """
+        super().__init__(nside=nside, map_dist=map_dist)
+        num_pix = hp.nside2npix(nside)
+        self.num_layers = num_layers
+        self.layers = np.empty((num_layers, 3, num_pix))
+        if isinstance(map_layers, (str, Path)):
+            for i_layer in range(num_layers):
+                self.layers[i_layer, 3, :] = self.read_map(
+                    map_layers.format(layer=i_layer), field=(0, 1, 2), unit=unit_layers
+                )
+        else:
+            self.layers = u.Quantity(map_layers, unit_layers)
+
+        self.freq_ref = u.Quantity(freq_ref).to(u.GHz)
+
+        with u.set_enabled_equivalencies(u.cmb_equivalencies(self.freq_ref)):
+            self.layers <<= u.uK_RJ
+
+        self.mbb_temperature = np.empty((num_layers, num_pix))
+
+        if isinstance(map_mbb_index, (str, Path)):
+            self.mbb_index = np.empty((num_layers, num_pix))
+            for i_layer in range(num_layers):
+                self.mbb_index[i_layer] = self.read_map(
+                    map_mbb_index.format(layer=i_layer), unit=""
+                )
+        else:
+            self.mbb_index = u.Quantity(map_mbb_index, unit="")
+
+        if isinstance(map_mbb_temperature, (str, Path)):
+            self.mbb_temperature = np.empty((num_layers, num_pix))
+            for i_layer in range(num_layers):
+                self.mbb_temperature[i_layer] = self.read_map(
+                    map_mbb_temperature.format(layer=i_layer), unit=unit_mbb_temperature
+                )
+        else:
+            self.mbb_temperature = u.Quantity(
+                map_mbb_temperature, unit=unit_mbb_temperature
+            )
+
+        self.mbb_temperature <<= u.K
+        self.nside = int(nside)
+
+    @u.quantity_input
+    def get_emission(self, freqs: u.GHz, weights=None) -> u.uK_RJ:
+        freqs = utils.check_freq_input(freqs)
+        weights = utils.normalize_weights(freqs, weights)
+        outputs = get_emission_numba(
+            freqs,
+            weights,
+            self.layers.value,
+            self.freq_ref.value,
+            self.mbb_index.value,
+            self.mbb_temperature.value,
+        )
+        return outputs << u.uK_RJ
+
+
+@njit(parallel=True)
+def get_emission_numba(
+    freqs,
+    weights,
+    layers,
+    freq_ref,
+    mbb_index,
+    mbb_temperature,
+):
+    npix = layers.shape[-1]
+    output = np.zeros((3, npix), dtype=layers.dtype)
+    temp = np.zeros((3, npix), dtype=layers.dtype)
+
+    I, Q, U = 0, 1, 2
+    for i, (freq, weight) in enumerate(zip(freqs, weights)):
+        for i_layer in range(layers.shape[0]):
+            temp[:, :] = layers[i_layer, :, :]
+            # -2 because black body is in flux unit and not K_RJ
+            temp[I] *= (freq / freq_ref) ** (mbb_index[i_layer] - 2.0)
+            temp[I] *= blackbody_ratio(freq, freq_ref, mbb_temperature[i_layer])
+            freq_scaling_P = (freq / freq_ref) ** (
+                mbb_index[i_layer] - 2.0
```

Comment:
@brandonshensley PySM 2 had this convention of subtracting 2 from the index, should I remove it for this model?